storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java

```diff
@@ -383,11 +396,11 @@ public void clear() {
 
     // Visible for testing
     public List<EpochEntry> epochEntries() {
-        lock.writeLock().lock();
+        lock.readLock().lock();
```

Comment:
Nice!

Could we move this to a separate PR please since I think this is an obvious change that can be shipped earlier than this PR.

Reply:
Opened another PR https://github.com/apache/kafka/pull/13483 to fix this bug.